diff --git a/CMakeLists.txt b/CMakeLists.txt
index 31fb959..b784f95 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -178,7 +178,6 @@ if(EMSCRIPTEN)
     add_link_options(-fexceptions)
     add_link_options(-sDISABLE_EXCEPTION_CATCHING=0)
   endif()
-  set(MANIFOLD_PYBIND OFF)
   set(BUILD_SHARED_LIBS OFF)
 endif()
 
diff --git a/bindings/CMakeLists.txt b/bindings/CMakeLists.txt
index bf76144..99cf64b 100644
--- a/bindings/CMakeLists.txt
+++ b/bindings/CMakeLists.txt
@@ -20,7 +20,7 @@ if(MANIFOLD_CBIND)
   add_subdirectory(c)
 endif()
 
-if(NOT EMSCRIPTEN AND MANIFOLD_PYBIND)
+if(MANIFOLD_PYBIND)
   add_subdirectory(python)
 endif()
 
diff --git a/bindings/python/manifold3d.cpp b/bindings/python/manifold3d.cpp
index 2123a1b..32726a4 100644
--- a/bindings/python/manifold3d.cpp
+++ b/bindings/python/manifold3d.cpp
@@ -576,7 +576,8 @@ NB_MODULE(manifold3d, m) {
           [](const MeshGL& self) {
             return nb::ndarray<nb::numpy, const float, nb::c_contig>(
                 self.vertProperties.data(),
-                {self.vertProperties.size() / self.numProp, self.numProp},
+                {static_cast<size_t>(self.vertProperties.size() / self.numProp),
+                 static_cast<size_t>(self.numProp)},
                 nb::handle());
           },
           nb::rv_policy::reference_internal)
@@ -691,7 +692,8 @@ NB_MODULE(manifold3d, m) {
           [](const MeshGL64& self) {
             return nb::ndarray<nb::numpy, const double, nb::c_contig>(
                 self.vertProperties.data(),
-                {self.vertProperties.size() / self.numProp, self.numProp},
+                {static_cast<size_t>(self.vertProperties.size() / self.numProp),
+                 static_cast<size_t>(self.numProp)},
                 nb::handle());
           },
           nb::rv_policy::reference_internal)
diff --git a/cmake/manifoldDeps.cmake b/cmake/manifoldDeps.cmake
index 07aaf98..d72d7a0 100644
--- a/cmake/manifoldDeps.cmake
+++ b/cmake/manifoldDeps.cmake
@@ -193,7 +193,10 @@ if(MANIFOLD_PYBIND)
   else()
     find_package(Python COMPONENTS Interpreter Development.SABIModule REQUIRED)
   endif()
-  if(Python_VERSION VERSION_GREATER_EQUAL 3.11)
+  if(CMAKE_CROSSCOMPILING)
+    set(MANIFOLD_PYBIND_STUBGEN OFF)
+    message("Cross-compiling, Python stub generation disabled")
+  elseif(Python_VERSION VERSION_GREATER_EQUAL 3.11)
     set(MANIFOLD_PYBIND_STUBGEN ON)
   else()
     # stubgen does not support version less than 3.11
